### ISSUE TITLE
Add half-integer scales to "Select Scale" menu on Retina

### ISSUE THREAD

**Issue by KelseyHigham** [OpenEmu/OpenEmu#2442]
<img width="206" alt="screen shot 2016-02-23 at 6 50 02 pm" src="https://cloud.githubusercontent.com/assets/286211/13274171/4c766972-da5e-11e5-93d6-67e44c7dc33e.png">
Right now, the player can select 1x and 2x (and so on). They should be able to select 0.5x and 1.5x.

In particular, 1.5x is the only resolution at which the included LCD3x filter displays correctly:
<img width="481" alt="screen shot 2016-02-23 at 6 52 38 pm" src="https://cloud.githubusercontent.com/assets/286211/13274225/a445b2de-da5e-11e5-8172-b97c761b614d.png">

**Comment by KelseyHigham** [comment]
Alternately, integer scales should be described in terms of physical pixels. So instead of 
- 0.5x
- 1x
- 1.5x
- 2x

the menu would read 
- 1x
- 2x
- 3x
- 4x

This would be consistent with the "LCD3x" terminology, anyway.

**Comment by clobber** [comment]
No, it doesn't make sense to do this. Especially if it's to only accommodate one shader (which you are suggesting to replace anyway, we get it!).

**Comment by KelseyHigham** [comment]
Sorry, I should've been clearer. The shader is why I noticed the issue, but it's not the real issue.

The "Select Scale" menu is intended to offer a shortcut to scale to _all_ integer sizes, but right now, it only offers a shortcut to _half_ of them.

Say you're playing a 480x272 PSP game on a 13" Retina MacBook, with 2560x1600 physical pixels. 

![openemu resolution feature request](https://cloud.githubusercontent.com/assets/286211/13279293/852a7452-da8b-11e5-9e5c-75025ba87473.png)

The resolutions that OpenEmu offers are green. The resolutions I'm suggesting are red. In this example, the largest resolution possible isn't available under the menu.

**Comment by mrvacbob** [comment]
I think it's valuable to support filters like LCD3x/HQ3x well since HQ4x is so slow it can hurt the framerate.

**Comment by mrvacbob** [comment]
Not sure if we ship hq3x but it definitely exists.

**Comment by clobber** [comment]
We're dropping Cg eventually anyway! It doesn't make much sense to work on these.

Also integral scaling isn't really about "resolution" so this half increments also doesn't make sense, especially if you are trying to base this around the resolution of just one system. Typically shaders should be written to be resolution independent.

**Comment by KelseyHigham** [comment]
How do half increments not make sense? They're half _points_, but they're whole _pixels_. 

What's the "Select Scale" option for, if not for quickly resizing to all available integer scales for the screen you're using?

**Comment by clobber** [comment]
Half increments doesn't make sense because this is *integer scaling*, whole numbers. You seem to be wanting to solve a different problem based on resolution. Many shaders (HQx, for example) are written in mind that the user should also be scaling the framebuffer at X or another whole number.

If all shaders are wrong on retina screens, then this is a problem to solve in the rendering step and not with the scaling factor options. To answer your question, the "Select Scale" option is to resize the scale of the game view to a whole number (a feature present in any emulator), not a particular resolution.

The subject of these particular Cg shaders in general is kind of pointless, though, as they will be dropped for GLSL (and in the future, Metal). The Cg spec has been dead for 4 years now and we won't continue to support these. On this platform, it's a wonder that the framework is still working at all.

**Comment by KelseyHigham** [comment]
On a Retina screen, half increments are integer sizes. Whole increments only represent even integers.

It's confusing to refer to them as half increments, because they're half-size logical _points_, but they're full-size physical _pixels_.

<img src="https://cloud.githubusercontent.com/assets/286211/13309546/9ac0b630-db32-11e5-9e37-6784c001fda3.png" width=363>

In this image, a 3x3 flower is shown at integer points on non-Retina, and at half-integer points on Retina. 

At the same time, it's integer _pixels_ on non-Retina, and integer _pixels_ on Retina.

Does that make sense?
